generic_config_updater/generic_updater.py

```diff
@@ -18,9 +21,94 @@ def release_lock(self):
         pass
 
 class ChangeApplier:
-    def apply(self, change):
-        # TODO: Implement change applier
-        raise NotImplementedError("ChangeApplier.apply(change) is not implemented yet")
+    def __init__(self, service_metadata_file: str=None):
+        self.config_db = ConfigDBConnector()
+        self.config_db.connect()
+        service_metadata = {}
+        if service_metadata_file:
+            with open(service_metadata_file, 'r') as f:
+                service_metadata = json.load(f)
+        self.table_service = service_metadata.get('tables', {})
+        self.service_commands = service_metadata.get('services', {})
+        self.valiate_metadata()
+        self.previous_table_name = None
+
+    def valiate_metadata(self):
+        # TODO: check metadata file syntax
+        pass
+
+    def exec_command(self, cmd) -> int:
+        p = subprocess.Popen(cmd)
+        p.communicate()
+        return p.returncode
+
+    def on_table_operation(self, table_name):
+        services = self.table_service.get(table_name, {})
+        for service in services:
+            commands = self.service_commands.get(service, {})
+            restart_comand = commands.get('restart-command')
+            validate_command = commands.get('validate-commands')
+            if restart_comand:
+                rc = self.exec_command(restart_comand)
+                if rc != 0:
+                    raise GenericConfigUpdaterError(f"Restart command failed: {restart_comand}, rc={rc}")
+                rc = self.exec_command(validate_command)
+                if rc != 0:
+                    raise GenericConfigUpdaterError(f"Validate command failed: {validate_command}, rc={rc}")
+
+    def on_table_operation_lazy(self, table_name):
+        # Optimze: reduce the duplicated restarting in a batch
+        if table_name != self.previous_table_name:
+            self.on_table_operation(self.previous_table_name)
+            self.previous_table_name = table_name
+
+    def _apply_entry(self, op: PatchOperation):
+        parts = op.pointer.parts
+        entry = self.config_db.get_entry(parts[0], parts[1])
+        oldtree = entry
+        for part in parts[1::-1]:
+            oldtree = { part: oldtree }
+        op.apply(oldtree)
+        self.config_db.set_entry(parts[0], parts[1], entry)
+
+    def apply(self, change: JsonChange):
+        patch = change.patch
```

Comment:
Using the patch within the change is ok, but it is limiting and is a little different than the design document. Using patch will forces specific order of execution on the `ChangeApplier` while in fact there is no ordering. This causes problems like multiple restarts which is solved by the lazy restart.

`JsonChange` on the other hand can be used to just getting the target config, and then the diff. The diff is per table, so we can handle table by table alphabetically.

`JsonChange` only provides one function which is apply (notice the second step in the [diagram](https://github.com/Azure/SONiC/blob/master/doc/config-generic-update-rollback/Json_Change_Application_Design.md#31-overview)) **GenerateTargetConfig**. I think `patch` access should be reduced i.e. to be preceded by `_`